### DISCUSSION TITLE
[Prettier Plugin] Prettier plugin not working with vs code and nextjs project

### DISCUSSION THREAD

**Discussion by AyushKoul00** [tailwindlabs/tailwindcss#11567] [Help]
I have created a next js project using create-next-app using npm in vs code. I have the prettier extension in vs code and my code is formatting properly (however, the class orders isn't changing). Everything else is working in prettier.

I have tried a lot of things:
- I have made the default formatter in vs code to prettier
- I have changed the `.prettierrc.json` file to `prettier.config.js` and `.prettierrc.cjs`
- I have tried to reload the vs code editor
- I have tried to uninstall and then install prettier v2.8 based on [this](https://github.com/tailwindlabs/prettier-plugin-tailwindcss/issues/176)
- I have checked "format on save" in the vs code settings
- I have run `npx prettier --write src/app/page.tsx ` (this does the same thing - formats the whole document but doesn't sort the class order. Everything else like tabs, spaces, etc is formatted)

My Info:
Editor: VS Code
OS: Windows 11
Project: next js + tailwind (next 13)
Prettier version: tried with 3 and 2.8
Package manage: npm

**Comment by kenlck** [comment]
I faced the same problem too. But i managed to get the plugin to work.
I had to set the prettier version to 2.8 and set some configuration in the prettier config file.

This is my prettier config file that make it works.

`
module.exports = {
  plugins: ["prettier-plugin-tailwindcss"],
  pluginSearchDirs: false,
};
`

After i added the prettier config i had to restart my VS code and the prettier plugin will start working.
